vm/src/hint_processor/builtin_hint_processor/bootloader/bootloader_hints.rs

```diff
@@ -832,4 +859,58 @@ mod tests {
         exec_scopes.insert_value(vars::PACKED_OUTPUT, composite_packed_output);
         assert!(assert_is_composite_packed_output(&mut exec_scopes).is_ok());
     }
+
+    #[rstest]
+    #[case(false)]
+    #[case(true)]
+    fn test_assert_program_address(#[case] expect_fail: bool) {
+        let mut vm = vm!();
+
+        add_segments!(vm, 2);
+        vm.run_context.fp = 2;
+
+        let ids_data = ids_data!("program_address");
+
+        let ap_tracking = ApTracking::new();
+
+        let mut new_segment_base = vm.add_memory_segment();
+        let _ = insert_value_from_var_name(
+            "program_address",
+            new_segment_base.clone(),
+            &mut vm,
+            &ids_data,
+            &ap_tracking,
+        )
+        .map_err(|e| panic!("could not insert var: {}", e));
```

Comment:
You don't really need to add a new segment. You can just insert an arbitrary relocatable value in `ids.program_address` and `exec_scopes` without having an actual address backing it up (i.e. the pointer does not need to be valid, nor point to a new segment). If you want a valid pointer, you can also just point to some address in the already created segments, like (0, 0) or (1, 0).

Reply:
Good point. Fixed in dae546b